### PR TITLE
Fix QUIC

### PR DESCRIPTION
**Pull request by wanghaEMQ**
Fix the conflict between msquic and nng close.
Fix some memory leakages in quic.
Fix deadlock in quic stream.
Fix the missing decrement of refcnt of dialer.
Fix the wrong nng_stream_free in quic_dialer_close.

### CODE REVIEW

**Review by JaylinYu — APPROVED** [review]

**Review by RanMaoyi — APPROVED** [review]

**Comment by JaylinYu** [comment]
Need to have a look on CI

**Comment by wanghaEMQ** [comment]
Which fixed all issues in https://github.com/nanomq/NanoNNG/issues/686

File: src/mqtt/protocol/mqtt/mqtt_quic_client.c [hunk 1]
```diff
@@ -332,6 +332,7 @@ mqtt_send_msg(nni_aio *aio, nni_msg *msg, mqtt_sock_t *s)
 	ptype = nni_mqtt_msg_get_packet_type(msg);
 	switch (ptype) {
 	case NNG_MQTT_CONNECT:
+	case NNG_MQTT_DISCONNECT:
```
[the remark this answers is not in the document]
Comment (wanghaEMQ) [reply src/mqtt/protocol/mqtt/mqtt_quic_client.c]: Yes. I have moved it down.

**Comment by wanghaEMQ** [comment]
> Need to have a look on CI

My mistake on codec

File: src/supplemental/quic/msquic_dial.c [hunk 2]
```diff
@@ -362,7 +361,11 @@ nni_quic_dialer_close(void *arg)
 				c->dial_aio = NULL;
 				nni_aio_set_prov_data(aio, NULL);
 				nng_stream_close(&c->stream);
-				nng_stream_free(&c->stream);
```
[the remark this answers is not in the document]
Comment (wanghaEMQ) [reply src/supplemental/quic/msquic_dial.c:365]: Yes. But I find I am wrong at later. I thought QUIC_STREAM_EVENT_SHUTDOWN_COMPLETE would be triggered only after the stream was established.

**Review by OdyWayne — APPROVED** [review]

**Review by lee-emqx — APPROVED** [review]

**Pull request merged**